src/cswinrt/strings/additions/Windows.Storage.Streams/WindowsRuntimeBufferExtensions.cs

```diff
@@ -639,17 +641,21 @@ public override void WriteByte(byte value)
                 _sourceBuffer.Length = (uint)Length;
             }
         }  // class WindowsRuntimeBufferUnmanagedMemoryStream
-
+
         private static Span<byte> GetSpanForCapacityUnsafe(this IBuffer buffer, uint offset)
         {
             Debug.Assert(0 <= offset);
             Debug.Assert(offset < buffer.Capacity);
 
-            unsafe
+            if (!WindowsRuntimeMarshal.TryGetDataUnsafe(buffer, out IntPtr buffPtr))

```

Comment:
Technically a GC hole (the JIT might reorder that property access), add a `GC.KeepAlive`.